### DISCUSSION TITLE
Converting 3D world coordinates into 2D view coordinates

### DISCUSSION THREAD

**Discussion by boeboe26** [pyvista/pyvista#4416] [Q&A] [answered]
Hi everyone!

I want to keep track of the x/y coordinates of some clicked points on the mesh.

In VTK, there's the function: 
virtual vtkMatrix4x4* GetCompositeProjectionTransformMatrix(double aspect, double nearz, double farz);

It converts 3D world coordinates to 2D viewport coordinates.

How do I do the same thing in Pyvista?

**Comment by tkoyama010** [comment]
Since PyVista has a Camera object that inherits from the vtkCamera object, you can call the method like `camera.GetCompositeProjectionTransformMatrix(aspect, nearz, farz)`.

**Comment by boeboe26** [comment, marked as answer]
Thank you for your answer, it helped a lot to find even more VTK functions that can be imported.

Here's how I obtained the 2D view coordinates without projecting the 3D points on my own:

```           
from vtkmodules.vtkRenderingCore import vtkCoordinate
    coordinate = vtkCoordinate()
    coordinate.SetCoordinateSystemToWorld()
    coordinate.SetValue(model_point_orig[0],model_point_orig[1],model_point_orig[2])

    viewCoord = coordinate.GetComputedDisplayValue(self.plotter.renderer)
```

Inspired by
https://stackoverflow.com/questions/76242749/converting-image-corner-coordinates-to-world-coordinates
